### PR TITLE
Add errors UI

### PR DESCRIPTION
**Pull request by Bookcliff**
I used SWR to manage the abi request and identify the errors.

Testing:

- [x] used invalid address string (0x75d078248eE49c585b73E73ab08bb3eaF93383A)
- [x] tested invalid contract address (0xd4c244ce2eb12fbf76e980379eb5480d6f99376f)
- [x] Reviewed etherscan's api docs to confirm that the additional errors they send are unrelated (https://docs.etherscan.io/support/common-error-messages)

I simplified the loading state using SWR.

Testing:

- [x] used valid contract address (0x75d078248eE49c585b73E73ab08bb3eaF93383Ae) to check loading state
- [x] used invalid contract address (0x75d078248eE49c585b73E73ab08bb3eaF93383A) to check that the loading and error states didn't break

### CODE REVIEW

**Review by RusseII — COMMENTED** [review]
Nice changes! Seems to function well. Is it possible to simplify the loading state? Here's an example of how they handle the loading state in the docs: https://swr.vercel.app/docs/getting-started#make-it-reusable

**Comment by Bookcliff** [comment]
> Nice changes! Seems to function well. Is it possible to simplify the loading state? Here's an example of how they handle the loading state in the docs: https://swr.vercel.app/docs/getting-started#make-it-reusable

Added!

File: src/App.js [hunk 1]
```diff
@@ -16,18 +16,28 @@ import { GithubOutlined } from "@ant-design/icons";
 
 const { Header, Footer, Content } = Layout;
 
+const fetcher = (...args) => fetch(...args).then((res) => res.json());
+
+function GetData(contractAddress) {
```
Comment (RusseII) [inline src/App.js:21]: this is a hook, so should be called `useData` 
CleanShot 2022-09-08 at 07 40 23@2x

https://reactjs.org/docs/hooks-overview.html#building-your-own-hooks

File: src/App.js [hunk 2]
```diff
@@ -73,34 +83,31 @@ function App() {
...
 
       const getError = () => {
-        if (error) {
+        if (isError) {
           setAbiError(true);
-          setLoading(false);
         } else if (
-          (data.abi === "Invalid Address format") |
-          (data.abi === "Contract source code not verified")
+          (contractEvents.abi === "Invalid Address format") |
```
Comment (RusseII) [inline src/App.js:95]: this is OK for now but you could simplify this by just throwing an error in the fetcher and you could remove the extra state variable https://swr.vercel.app/docs/error-handling

**Review by RusseII — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Bookcliff) [reply src/App.js:21]: Done

File: src/App.js (hunk 2 above)
Comment (Bookcliff) [reply src/App.js]: Done

**Review by RusseII — APPROVED** [review]

**Pull request merged**